packages/utils/src/abi_decoder.ts

```diff
@@ -89,25 +110,94 @@ export class AbiDecoder {
         }
     }
     /**
-     * Add additional ABI definitions to the AbiDecoder
-     * @param abiArray An array of ABI definitions to add to the AbiDecoder
+     * Decodes transaction data for a known ABI.
+     * @param calldata hex-encoded transaction data.
+     * @param contractName used to disambiguate similar ABI's (optional).
+     * @return Decoded transaction data. Includes: function name and signature, along with the decoded arguments.
      */
-    public addABI(abiArray: AbiDefinition[]): void {
+    public tryDecodeCalldata(calldata: string, contractName?: string): TransactionData {
+        const functionSelector = AbiDecoder._getFunctionSelector(calldata);
+        const candidateFunctionInfos = this._functionInfoBySelector[functionSelector];
+        if (_.isUndefined(candidateFunctionInfos)) {
+            throw new Error(`No functions registered for selector '${functionSelector}'`);
+        }
+        const functionInfo = _.find(candidateFunctionInfos, txDecoder => {
+            return _.isUndefined(contractName) || _.toLower(txDecoder.contractName) === _.toLower(contractName);
+        });
+        if (_.isUndefined(functionInfo)) {
+            throw new Error(
+                `No function registered with selector ${functionSelector} and contract name ${contractName}.`,
+            );
+        } else if (_.isUndefined(functionInfo.abiEncoder)) {
+            throw new Error(
+                `Function ABI Encoder is not defined, for function registered with selector ${functionSelector} and contract name ${contractName}.`,
+            );
+        }
+        const functionName = functionInfo.abiEncoder.getDataItem().name;
+        const functionSignature = functionInfo.abiEncoder.getSignatureType();
+        const functionArguments = functionInfo.abiEncoder.decode(calldata);
+        const decodedCalldata = {
+            functionName,
+            functionSignature,
+            functionArguments,
+        };
+        return decodedCalldata;
+    }
+    /**
+     * Adds a set of ABI definitions, after which transaction data targeting these ABI's can be decoded.
+     * Additional properties can be included to disambiguate similar ABI's. For example, if two functions
+     * have the same signature but different parameter names, then their ABI definitions can be disambiguated
+     * by specifying a contract name.
+     * @param abiDefinitions ABI definitions for a given contract.
+     * @param contractName Name of contract that encapsulates the ABI definitions (optional).
+     */
+    public addABI(abiArray: AbiDefinition[], contractName?: string): void {
         if (_.isUndefined(abiArray)) {
             return;
         }
         const ethersInterface = new ethers.utils.Interface(abiArray);
```

Comment:
Should we drop ethers here?

Reply:
See above. This will be in a subsequent PR.